### PR TITLE
Moved sending new follower notification to a separate job

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
- created a separate Active Job to handle new follower notification
- pass attributes instead of a `Follow` instance to the job to get rid of deserialization error.
(in this case, we can't just pass `follow_id`, cause a `follow` might be destroyed at that moment)
- moved the new follower notification logic to a service object
- kept the old `Notification` interface (`send_new_follower_notification` and `send_new_follower_notification_without_delay`)

## Related Tickets & Documents
#1996

### CODE REVIEW

**Comment by lightalloy** [comment]
Noticed another notifications issue:
I suppose that notifications should only be sent if followable_type is `User` or `Organization`.
https://github.com/thepracticaldev/dev.to/blob/master/app/controllers/follows_controller.rb#L27-L35
Otherwise, an incorrect notification could be updated or attempted to be created:
https://github.com/thepracticaldev/dev.to/blob/master/app/models/notification.rb#L30
If a tag follow is passed the params here will be `{ action: "Follow" }`

File: app/services/notifications/new_follower/follow_data.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require "dry-types"
+require "dry-struct"
+
+module Notifications
+  module NewFollower
+    module Types
+      include Dry::Types.module
+    end
+
+    class FollowData < Dry::Struct
```
Comment (lightalloy) [inline app/services/notifications/new_follower/follow_data.rb]: I used `dry-struct` to check the input passed to the service to prevent errors like #2041. This gem can also be useful in the app when we need to check structure.

Comment (maestromac) [reply app/services/notifications/new_follower/follow_data.rb]: Very clean. Love this.

File: Gemfile.lock [hunk 2]
```diff
@@ -1085,4 +1151,4 @@ RUBY VERSION
    ruby 2.6.1p33
 
 BUNDLED WITH
-   1.17.3
+   2.0.1
```
Comment (maestromac) [inline Gemfile.lock]: I'm not sure if you intentionally updated bundler along with some dependencies but I'm having trouble running the app locally now (something with `push-notifications-ruby` is messing me up). Heroku also does not support Bundler 2.0 without additional configuration. Can you please revert these changes.

Comment (lightalloy) [reply Gemfile.lock]: Oops, that was not intentional. Probably I messed this up while rebasing. Will change back.
